wappsto/connection/communication.py

```diff
@@ -623,23 +626,31 @@ def receive_data(self):
             The decoded message from the socket.
 
         """
-        total_decoded = []
+        total_decoded = ''
         decoded = None
         while True:
             if self.connected:
-                data = self.my_socket.recv(2000)
-                decoded_data = data.decode('utf-8')
-                total_decoded.append(decoded_data)
-            else:
-                break
-
-            try:
-                decoded = json.loads(''.join(total_decoded))
-            except JSONDecodeError:
+                data = self.my_socket.recv(RECEIVE_SIZE)
                 if data == b'':
                     self.reconnect()
+                try:
+                    decoded_data = data.decode('utf-8')
+                except:
+                    continue
+                total_decoded += decoded_data
+                if sys.getsizeof(total_decoded) > MESSAGE_SIZE_BYTES:
```

Comment:
Why not just use 'len'?

Reply:
len(): returns us the number of elements present in the sequence
getsizeof(): check how much memory in bytes is allocated to store a complete object

the example is shown here: https://www.happyprogrammingguide.com/wp-content/uploads/2019/05/Python-check-length-and-size-of-string.-1024x576.jpg